store/store_test.go

```diff
@@ -2582,6 +2582,54 @@ func (t *remoteRepoTestSuite) TestUbuntuStoreRepositoryListRefreshRetryOnEOF(c *
 	c.Assert(results[0].Name(), Equals, "hello-world")
 }
 
+func (t *remoteRepoTestSuite) TestUbuntuStoreUnexpectedEOFhandling(c *C) {
+	mockServerHelper := func(failures int) error {
+		var mockServer *httptest.Server
+		n := 0
+		mockServer = httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			n++
+			if n < failures {
```

Comment:
So we write the header for at most `failures` requests made. Why do you need the counter if you set it to a huge value?